lib/hotfixes.sh

```diff
@@ -0,0 +1,260 @@
+    # Assumes CHART_NAMES has been set to list of desired chart names
+    # Assumes YAML files exist for each chart in the target directory
+    # Assumes HOTFIX_LABEL has been set
+    # Stores resulting file location in $manifest_file
+    local chart_file chart_files chart
+
+    if [[ $# -ne 0 ]]; then
+        err_echo "$0 function accepts no arguments but received $#: $*"
+        return 1
+    elif [[ ${#CHART_NAMES[@]} -eq 0 ]]; then
+        err_echo "$0: CHART_NAMES array variable is empty"
+        return 1
+    elif [[ -z ${workdir} ]]; then
+        err_echo "$0: workdir variable not set"
+        return 1
+    elif [[ ! -e ${workdir} ]]; then
+        err_echo "$0: workdir directory does not exist: '${workdir}'"
+        return 1
+    elif [[ ! -d ${workdir} ]]; then
+        err_echo "$0: workdir exists but is not a directory: '${workdir}'"
+        return 1
+    fi
+    
+    chart_files=()
+    # Make sure all chart names are valid and the files exist
+    for chart in "${CHART_NAMES[@]}"; do
+        if [[ "${chart}" =~ ^[^a-z0-9] ]]; then
+            err_echo "$0: Invalid starting character in chart name: '${chart}'"
+            return 1
+        elif [[ "${chart}" =~ [^a-z0-9]$ ]]; then
+            err_echo "$0: Invalid final character in chart name: '${chart}'"
+            return 1
+        elif [[ "${chart}" =~ [^-a-z0-9] ]]; then
+            err_echo "$0: Invalid characters in chart name: '${chart}'"
+            return 1
+        fi
+        chart_file="${workdir}/${chart}.yaml"
+        if [[ ! -e ${chart_file} ]]; then
+            err_echo "$0: Chart file does not exist: '${chart_file}'"
+            return 1
+        elif [[ ! -f ${chart_file} ]]; then
+            err_echo "$0: Chart file exists but is not a regular file: '${chart_file}'"
+            return 1
+        elif [[ ! -r ${chart_file} ]]; then
+            err_echo "$0: Chart file not readable: '${chart_file}'"
+            return 1
+        elif [[ ! -w ${chart_file} ]]; then
+            err_echo "$0: Chart file not writable: '${chart_file}'"
+            return 1
+        fi
+        chart_files+=( "${chart_file}" )
+    done
+
+    # Create hotfix manifest_outline
+manifest_file="${workdir}/hotfix_manifest.yaml"
+cat <<EOF > "${manifest_file}"
+apiVersion: manifests/v1beta1
+metadata:
+  name: ${HOTFIX_LABEL}
+spec:
+  sources:
+    charts:
+      - location: https://packages.local/repository/charts
+        name: csm-algol60
+        type: repo
+EOF
+
+    # For each chart file, move it into manifest format.
+    # Hacky to do this with sed, but it's a very simple text modification.
+    #
+    # Basically convert it from:
+    # name: whatever
+    # version: whatever
+    # ...
+    #
+    # to:
+    #spec:
+    #  charts:
+    #    - name: whatever
+    #      version: whatever
+    #      ...
+    sed -i -e '1s/^/spec:\n  charts:\n    - /' -e '2,$s/^/      /' "${chart_files[@]}"
+    
+    # Merge them all into manifest.yaml
```

Comment:
```suggestion
    # Merge them all into manifest.yaml
    # --arrays=append will merge the "charts:" lists from the different chart files, instead of having them overrwrite each other
```